src/check/parse/Parser.zig

```diff
@@ -1334,6 +1333,25 @@ pub fn parsePattern(self: *Parser, alternatives: Alternatives) AST.Pattern.Idx {
     } });
 }
 
+fn parseAsPattern(self: *Parser, pattern: AST.Pattern.Idx) AST.Pattern.Idx {
+    if (self.peek() != .KwAs) {
+        return pattern;
+    }
+    self.advance(); // Advance past KwAs
+    if (self.peek() != .LowerIdent) {
+        // The name of a pattern can only be a lower ident
+        return self.pushMalformed(AST.Pattern.Idx, .bad_as_pattern_name, self.pos);
+    }
```

Comment:
Will this work for imports, not sure if that is handled elsewhere. `import Json as J`